### DISCUSSION TITLE
feature: support multicall2

### DISCUSSION THREAD

**Discussion by 0xCryptoZen** [wevm/viem#1203] [Idea / Feature Request]
Now just has multicall and multicall3, why don't support multicall2 ?

**Comment by tmm** [comment]
Where does multicall3 fall short and multicall2 is preferred?

**Reply by murderteeth** [reply]
multicall3 doesn't fall short, but it's not available for mainnet blocks before block 14353601, for example. could viem support a multicall2 fallback?

**Reply by robiiinos** [reply]
Read https://github.com/wagmi-dev/viem/discussions/183 and https://github.com/wagmi-dev/viem/pull/1443
